pkg/controller/node/controller_utils.go

```diff
@@ -250,3 +277,43 @@ func recordNodeStatusChange(recorder record.EventRecorder, node *api.Node, new_s
 	// and event is recorded or neither should happen, see issue #6055.
 	recorder.Eventf(ref, api.EventTypeNormal, new_status, "Node %s status is now: %s", node.Name, new_status)
 }
+
+type modifyTaintsFunc func(oldTaints []api.Taint) ([]api.Taint, bool, error)
+
+func tryModifyNodeTaints(kubeClient clientset.Interface, nodeName string, modifyTaints modifyTaintsFunc) (bool, error) {
+	var updateErr error
+	for attempt := 0; attempt < nodeStatusUpdateRetry; attempt++ {
+		node, err := kubeClient.Core().Nodes().Get(nodeName)
```

Comment:
To update taints, we need to patch the whole json encoded taints string, if don't get before retrying, would probably drop taints newly added by someone else.